utils/lib/Dashboard.ts

```diff
@@ -106,6 +145,65 @@ class Dashboard extends Component<DashboardConfig, QuickstartDashboardInput> {
     };
   }
 
+  /**
+   * Static method of data sources associated
+   * with dashboard template id
+   * @returns - object with ids and NGerrors
+   */
+  static async getRequiredDataSources(
+    templateId: string
+  ): Promise<{ ids: string[]; errors?: ErrorOrNerdGraphError[] }> {
+    const { data, errors } = await fetchNRGraphqlResults<
+      DashboardRequiredDataSourcesQueryVariables,
+      DashboardRequiredDataSourcesQueryResults
+    >({
+      queryString: DASHBOARD_REQUIRED_DATA_SOURCES_QUERY,
+      variables: { id: templateId },
+    });
+
+    if (errors) {
+      Promise.reject(`Query for dashboard template id failed`);
```

Comment:
If we reject the promise here, don't we never return the errors from query for the data sources? @MichelLosier 

So either we need to get rid of line 188-190 or if we only want to reject on the outer layer then we can get rid of this and instead just rely on the return of line 188-190

Reply:
I'm go ahead of getting rid of those lines and just rely on rejecting the promise here

Reply:
ya good call, went ahead and removed that promise rejection